cpp/src/Slice/Grammar.y

```diff
@@ -1890,15 +1881,42 @@ dictionary_def
 // ----------------------------------------------------------------------
 enum_id
 // ----------------------------------------------------------------------
-: ICE_ENUM ICE_IDENTIFIER
+: ICE_UNCHECKED ICE_ENUM ICE_IDENTIFIER
```

Comment:
Instead of having two separate sections for handling enums prefixed with `ICE_UNCHECKED` and those without, we could combine the into one rule and avoid duplication by moving the check for `ICE_UNCHECKED` into it's own match like so:
```
// ----------------------------------------------------------------------
unchecked_qualifier
// ----------------------------------------------------------------------
: ICE_UNCHECKED
{
    BoolTokPtr out = new BoolTok;
    out->v = true;
    $$ = out;
}
| %empty
{
    BoolTokPtr out = new BoolTok;
    out->v = false;
    $$ = out;
}
;
```

And then changing this rule to match
```
unchecked_qualifier ICE_ENUM ICE_IDENTIFIER
```
where the bool indicating whether it's unchecked or not could be accessed by `$1->v`

Reply:
Unfortunately this doesn't work.

Reply:
Unfortunately indeed... I think the alternative I sent you on Slack should work, but it doesn't look as clean as this one does, so it might not be worth the change...